neoarch/managers/docker_manager.py

```
@@ -629,13 +794,25 @@ def show_container_menu(self, pos):
+        term = None
+        for t in terminals:
+            if shutil.which(t):
+                term = t
+                break
+        for shell in ["/bin/bash", "/bin/sh"]:
             try:
-                subprocess.Popen(["docker", "exec", "-it", cid, "/bin/sh"])
-            except Exception as e:
-                self.log_signal.emit(f"Failed to open shell: {e}")
+                if term == "gnome-terminal":
+                    subprocess.Popen([term, "--", "bash", "-c",
```

Ruff (0.15.15)

[error] `subprocess` call: check for execution of untrusted input

(S603)